phylanx/ir/node_data.hpp

```diff
@@ -370,6 +371,17 @@ namespace phylanx { namespace ir
             return *this;
         }
 
+        template <typename U,
+            typename U1 =
+                typename std::enable_if<!std::is_same<T, U>::value>::type>
+        node_data& operator=(node_data const& d)
```

Comment:
Shouldn't this be:

    node_data& operator=(node_data<U> const& d) {...}

?